packages/api/src/index.ts

```diff
@@ -0,0 +1,545 @@
+      log.error(error, 'Error while getting user activity')
+      res.status(500).json({ error: 'Internal Server Error' })
+      return
+    }
+  })
+)
+
+const workerHandler = awrap(async function (
+  req: Request,
+  res: Response
+): Promise<void> {
+  const body = req.body as {
+    task?: string
+    full?: boolean
+    contractAddresses?: string[]
+    requestLimit?: number
+  }
+
+  log.info(
+    {
+      contractAddresses: body.contractAddresses,
+      full: body.full,
+      requestLimit: body.requestLimit,
+    },
+    `Received task: ${body.task ?? 'default'}`
+  )
+
+  if (body.task === 'wallet') {
+    try {
+      await updateWallets()
+    } catch (err) {
+      // console.error(err)
+      log.error(err, 'Error updating wallets')
+      res.status(500).json({ success: false, message: 'Internal server error' })
+      return
+    }
+  } else {
+    const fullHistory = body.full === true
+    const requestLimit = body.requestLimit ?? 10
+    const contractAddresses = (body.contractAddresses ?? [])
+      .map(addressMaybe)
+      .filter((x) => x)
+
+    try {
+      for (const contractAddress of contractAddresses) {
+        await collectActivities({
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          contractAddress: contractAddress!,
+          fullHistory,
+          requestLimit,
+        })
+      }
+    } catch (err) {
+      // console.error(err)
+      log.error(err, 'Error fetching listings in fetchHandler')
+      res.status(500).json({ success: false, message: 'Internal server error' })
+      return
+    }
+  }
+
+  res.status(200).json({ success: true })
+})
+
+if (isWorker) {
+  app.post('/', workerHandler)
+} else if (!isProdEnv) {
+  // Conditional local runner
+  app.post('/work', workerHandler)
+}
+
+// Add a collection
+app.post(
+  '/collection',
+  apiKeyMiddleware,
+  awrap(async function (req: Request, res: Response): Promise<void> {
+    const body = req.body as {
+      contractAddress: string
+      description?: string
+      disabled?: boolean
+    }
+    const contractAddress = addressMaybe(body.contractAddress)
+    const { disabled = false, description = '' } = body
+
+    if (!contractAddress) {
+      const msg = 'Invalid contractAddress'
+      log.warn({ contractAddress }, msg)
+      res.status(400).json({ error: msg })
+      return
+    }
+
+    const props = {
+      description,
+      disabled,
+    }
+    const [collection, created] = await Collection.findOrCreate({
+      where: { contractAddress: hex2buf(contractAddress) },
+      defaults: {
+        ...props,
+        contractAddress: hex2buf(contractAddress),
+      },
+    })
+
+    if (!created) {
+      await collection.update(props)
+    }
+
+    res.status(200).json({
+      success: true,
+      message: created ? 'Created collection' : 'Updated collection',
+    })
+  })
+)
+
+// Trigger a fetch sequence
+// TODO: Do we want this exposed in prod?  Periodic beanstalk task?
```

Comment:
I could not figure what calls /trigger? Is it using a cron on the beanstalk worker tier? like https://repost.aws/knowledge-center/elastic-beanstalk-cron-job-worker-tier
If yes, I did not see the config for it.

Reply:
Currently a cron on an ec2 instance...